Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1525,7 +1525,8 @@ void WebPageProxy::finishAttachingToWebProcess(const Site& site, ProcessLaunchRe
     if (reason != ProcessLaunchReason::ProcessSwap)
         initializeWebPage(site, m_mainFrame ? m_mainFrame->effectiveSandboxFlags() : configuration().initialSandboxFlags());
 
-    protectedInspector()->updateForNewPageProcess(*this);
+    if (RefPtr inspector = this->inspector())
```

Comment:
This is fine but I think that also:
1. We want to early return in `WebPageProxy::loadAndDecodeImage()` when `isClosed()`. It makes no sense to try and launch a process after the page has been closed
2. It would be good to write an API test